src/context/StationsContext.js

```diff
@@ -0,0 +1,58 @@
+import { createContext, useEffect, useState } from 'react';
+import Cookies from 'js-cookie';
+
+import { setBasicRadioStationsToLS } from '../utils/setBasicRadioStations';
+export const StationsContext = createContext();
+
+export const StationsContextProvider = ({ children }) => {
+  const [isModalOpen, setModalIsOpen] = useState(false);
+  const [radioStationsListLS, setRadioStationsListLS] = useState([]);
+
+  useEffect(() => {
+    if (!Cookies.get('visited')) {
+      Cookies.set('visited', true);
+      // set basic radio stations in local storage on first visit
+      setBasicRadioStationsToLS();
+    }
+    refreshRadioStations();
+  }, []);
+
+  function refreshRadioStations() {
+    let radioStations = JSON.parse(localStorage.getItem('stations'));
+    setRadioStationsListLS(radioStations);
+  }
+
+  function addNewRadioStationToLS(data, setModalIsOpen, radioStationsListLS) {
```

Comment:
This function's name shouldn't contain information about what the data is being saved to. This one could be named `saveRadioStation` or just `addNewRadioStation` - you abstract things out so that you won't need to care about the technicalities underneath :)

Also you could use closure and use the `radioStationsListLS` from your hook's lexical scope